src/shared/conf-parser.c

```diff
@@ -679,6 +678,45 @@ int config_parse_many(
         return 0;
 }
 
+int config_parse_standard_file_with_dropins_full(
+                const char *root,
+                const char *main_file,    /* A path like "systemd/frobnicator.conf" */
+                const char *sections,
+                ConfigItemLookup lookup,
+                const void *table,
+                ConfigParseFlags flags,
+                void *userdata,
+                Hashmap **ret_stats_by_path,
+                char ***ret_dropin_files) {
+
+        char **conf_paths = CONF_PATHS_STRV("");
+        _cleanup_strv_free_ char **configs = NULL;
+        int r;
+
+        /* Build the list of main config files */
+        r = strv_extend_strv_prepend_append(&configs, root, conf_paths, main_file);
+        if (r < 0) {
+                if (flags & CONFIG_PARSE_WARN)
+                        log_oom();
+                return r;
+        }
+
+        const char *dropin_dirname = strjoina(main_file, ".d");
+
+        return config_parse_many(
+                        (const char* const*) configs,
+                        (const char* const*) conf_paths,
```

Comment:
Instead of doing the cast here, declare `conf_paths` as `const char* const*` and do the cast in `strv_extend_strv_biconcat` sounds more correct?

Reply:
I did that now. It requires adding a bunch of casts, but I think it's more correct to do it this way.